libraries/SRV_Channel/SRV_Channels.cpp

```diff
@@ -221,6 +230,23 @@ void SRV_Channels::set_output_pwm_chan(uint8_t chan, uint16_t value)
     }
 }
 
+// set output value for a specific function channel as a pwm value with loop based timeout
+// timeout_ms of zero will clear override of the channel
+// minimum override is 1 MAIN_LOOP
+void SRV_Channels::set_output_pwm_chan_timeout_ms(uint8_t chan, uint16_t value, uint16_t timeout_ms)
+{
+    WITH_SEMAPHORE(_singleton->calc_sem);
+
+    if (chan < NUM_SERVO_CHANNELS) {
+        if (timeout_ms == 0) {
+            override_counter[chan] = 0;
+        } else {
+            override_counter[chan] = timeout_ms * 1000 / AP::scheduler().get_loop_period_us() + 1;
```

Comment:
const uint32_t loop_period_us = AP::scheduler().get_loop_period_us();
override_counter[chan] = (timeout_ms * 1000U + loop_period_ms-1) / AP::scheduler().get_loop_period_us()

Reply:
After many send_text messages in the terminal, I can confirm that this does the expected behavior of set override_counter to input ms time with a minimum time of 1 loop period.